[AOTI] Update test runner to use the new APIs

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #147105

Summary: Switch to the newer aoti_compile_and_package APIs. Some tests still kept using legacy APIs, and will follow up with internal test refactoring.

cc @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

Differential Revision: [D69609685](https://our.internmc.facebook.com/intern/diff/D69609685)

desertfire committed:
Update

[ghstack-poisoned]

jingsh commented on test/inductor/test_aot_inductor_utils.py:
```diff
@@ -30,13 +30,16 @@ def forward(self, *args, **kwargs):
         dynamic_shapes=None,
         disable_constraint_solver=False,
     ):
+        if not isinstance(model, torch.nn.Module):
+            model = WrapperModule(model)
+
```
duplicated? already exist

jingsh reviewed (approved):
lgtm

desertfire committed:
Update

[ghstack-poisoned]

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D69609685).

desertfire committed:
Update

[ghstack-poisoned]

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D69609685).

desertfire committed:
Update

[ghstack-poisoned]

desertfire committed:
Update on "[AOTI] Update test runner to use the new APIs"


Summary: Switch to the newer aoti_compile_and_package APIs. Some tests still kept using legacy APIs, and will follow up with internal test refactoring.

cc H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D69609685](https://our.internmc.facebook.com/intern/diff/D69609685)

[ghstack-poisoned]

desertfire committed:
Update on "[AOTI] Update test runner to use the new APIs"


Summary: Switch to the newer aoti_compile_and_package APIs. Some tests still kept using legacy APIs, and will follow up with internal test refactoring.

cc H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D69609685](https://our.internmc.facebook.com/intern/diff/D69609685)

[ghstack-poisoned]

desertfire committed:
Update on "[AOTI] Update test runner to use the new APIs"


Summary: Switch to the newer aoti_compile_and_package APIs. Some tests still kept using legacy APIs, and will follow up with internal test refactoring.

cc H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D69609685](https://our.internmc.facebook.com/intern/diff/D69609685)

[ghstack-poisoned]

desertfire committed:
Update on "[AOTI] Update test runner to use the new APIs"


Summary: Switch to the newer aoti_compile_and_package APIs. Some tests still kept using legacy APIs, and will follow up with internal test refactoring.

cc H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D69609685](https://our.internmc.facebook.com/intern/diff/D69609685)

[ghstack-poisoned]

desertfire committed:
Update on "[AOTI] Update test runner to use the new APIs"


Summary: Switch to the newer aoti_compile_and_package APIs. Some tests still kept using legacy APIs, and will follow up with internal test refactoring.

cc H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D69609685](https://our.internmc.facebook.com/intern/diff/D69609685)

[ghstack-poisoned]

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D69609685).

desertfire commented:
Will revert once the internal unland finishes.
